Support Want-Digest for external content with Head and Get

lsitu opened the pull request:
Fixes https://jira.duraspace.org/browse/FCREPO-2587

Add support for Head and Get with Want-Digest header to external content.

awoods commented on fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/impl/CacheEntryFactory.java:
```diff
@@ -44,11 +53,29 @@ private CacheEntryFactory() {
+    /**
+     * Utility method to parse the external body content in format:
+     *   message/external-body; access-type=URL; url="http://www.example.com/file"
+     * @param mimeType
```
```
[WARNING] /fcrepo4/fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/impl/CacheEntryFactory.java:71: warning: no description for @param
[WARNING] * @param mimeType
```

awoods commented on fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraBinaryImpl.java:
```diff
@@ -383,15 +384,33 @@ public RdfStream getFixity(final IdentifierConverter<Resource, FedoraResource> i
+
+                    return CacheEntryFactory.forProperty(getProperty(HAS_MIME_TYPE)).checkFixity(algorithms);
+                } else {
+                    throw new UnsupportedAccessTypeException("Unsupported access-type " + accessType
```
I would expect the following command (with JUNK access-type) to fail... but it succeeds:
```
curl -i -ufedoraAdmin:fedoraAdmin -XPUT -H"Content-Type: message/external-body;access-type=JUNK;url=\"https://blognumbers.files.wordpress.com/2010/09/6.jpg\"" localhost:8080/rest/image

```
I see it fails when requesting `Want-Digest`... but should also fail at the point of creation. That can be a new, follow-on ticket.
https://fcrepo.github.io/fcrepo-specification/#external-content

awoods replied:
Nevermind. I see this issue is addressed in: https://jira.duraspace.org/browse/FCREPO-2586

lsitu commented on fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/impl/CacheEntryFactory.java:
```diff
@@ -44,11 +53,29 @@ private CacheEntryFactory() {
+        final Map<String, String> params = new HashMap<String, String>();
+        Splitter.on(';').omitEmptyStrings().trimResults()
+            .withKeyValueSeparator(Splitter.on('=').limit(2)).split("mime-type=" + mimeType.trim())
+            .forEach((k, v) -> params.put(k.toLowerCase(), v.replaceAll("^\"|\"$", "")));
```
Sure. Comment added in commit https://github.com/fcrepo4/fcrepo4/pull/1238/commits/22c52a57b4d391e0bc1c30b871e0d1a3300d9bcb.

lsitu commented:
@awoods I've addressed the issues in commit https://github.com/fcrepo4/fcrepo4/pull/1238/commits/22c52a57b4d391e0bc1c30b871e0d1a3300d9bcb. And the PR is rebased.
